nessie/src/test/java/org/apache/iceberg/nessie/TestBranchVisibility.java

```diff
@@ -173,6 +174,31 @@ public void testSchemaSnapshot() throws Exception {
     Assertions.assertThat(metadataOn2).isNotEqualTo(metadataOnTest).isNotEqualTo(metadataOnTest2);
   }
 
+  @Test
+  public void testMetadataLocation() throws Exception {
+    String branch1 = "test";
+    String branch2 = "branch-2";
+
+    // commit on tableIdentifier1 on branch1
+    NessieCatalog catalog = initCatalog(branch1);
+    String metadataLocationOfCommit1 = addRow(catalog, tableIdentifier1, "initial-data",
+        ImmutableMap.of("id0", 4L));
+
+    createBranch(branch2, catalog.currentHash(), branch1);
+    // commit on tableIdentifier1 on branch2
+    catalog = initCatalog(branch2);
+    String metadataLocationOfCommit2 = addRow(catalog, tableIdentifier1, "some-more-data",
+        ImmutableMap.of("id0", 42L));
+    Assertions.assertThat(metadataLocationOfCommit2).isNotEqualTo(metadataLocationOfCommit1);
```

Comment:
nit: maybe change to `.isNotNull().isNotEqualTo(metadataLocationOfCommit1)`. same for the other check further below